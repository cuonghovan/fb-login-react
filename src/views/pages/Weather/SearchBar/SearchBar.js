import React from 'react';
import PropTypes from 'prop-types';

class SearchBar extends React.Component {
  constructor(props) {
    super(props);

    this.onSearchBtnClick = this.onSearchBtnClick.bind(this);
  }

  static propTypes = {
    handleSearch: PropTypes.func
  };

  onSearchBtnClick() {
    let query = this.searchInput.value.trim();
    this.props.handleSearch(query);
  }

  render() {
    return(
      <div className='search-bar'>
        <input type='text' tabIndex='0'placeholder='Enter location' ref={c => this.searchInput = c} />
        <button onClick={this.onSearchBtnClick}>Search</button>
      </div>
    );
  }
}

export default SearchBar;
