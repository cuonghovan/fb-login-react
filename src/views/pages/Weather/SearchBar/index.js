export { default as SearchBar } from './SearchBar';
export { default } from './SearchBarContainer';
