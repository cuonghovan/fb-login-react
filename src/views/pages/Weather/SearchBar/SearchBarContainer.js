import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import { fetchWeatherInfoAct } from "State/weather";
import SearchBar from "./SearchBar";

class SearchBarContainer extends React.Component {
  constructor(props) {
    super(props);

    this.handleSearch = this.handleSearch.bind(this);
  }

  static propTypes = {
    fetchWeatherInfoAct: PropTypes.func
  };

  handleSearch(query) {
    this.props.fetchWeatherInfoAct(query);
  }

  render() {
    return <SearchBar handleSearch={this.handleSearch} />;
  }
}

const mapDispatchToProps = {
  fetchWeatherInfoAct
};

export default connect(null, mapDispatchToProps)(SearchBarContainer);
