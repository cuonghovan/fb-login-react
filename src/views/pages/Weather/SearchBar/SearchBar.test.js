/*global shallow mount*/
import React from "react";
import "jest-styled-components";
import SearchBar from "./SearchBar";

describe("search-bar", () => {
  const mockHandleSearch = jest.fn();
  const props = {
    handleSearch: mockHandleSearch
  };

  it("renders correctly", () => {
    const searchBar = shallow(<SearchBar {...props} />);
    expect(searchBar).toMatchSnapshot();
  });

  describe("when user enters location and clicks search button", () => {
    const searchBar = mount(<SearchBar {...props} />);
    let queryStr = "hanoi";
    let input = searchBar.find("SearchBar__Input").find("input");
    let searchButton = searchBar.find("SearchBar__SearchButton");

    input.instance().value = queryStr;
    input.simulate("change");
    searchButton.simulate("click");

    it("dispatchs the `handleSearch()` method it receives from props", () => {
      expect(mockHandleSearch).toBeCalledWith(queryStr);
    });
  });
});
