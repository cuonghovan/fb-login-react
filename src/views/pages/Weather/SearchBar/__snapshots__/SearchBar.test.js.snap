// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search-bar renders correctly 1`] = `
<SearchBar__Wrapper>
  <SearchBar__Input
    innerRef={[Function]}
    placeholder="Enter location"
    tabIndex="0"
    type="text"
  />
  <SearchBar__SearchButton
    onClick={[Function]}
  >
    Search
  </SearchBar__SearchButton>
</SearchBar__Wrapper>
`;
